### ISSUE TITLE
bug: Opening an alert from form's handler blocks the UI

### ISSUE THREAD

**Issue by iignatov** [driftyco/ionic#5799]
#### Short description of the problem:
When opening an alert from inside of a form's submit handler and after opening/dismissing this alert the UI becomes inaccessible.

#### What behavior are you expecting?
The alert should disappear and the UI should remain accessible as when opening the alert from a button outside of the form.

**Steps to reproduce:**
1. Open: https://plnkr.co/edit/MxaYmv7jEeF5NHiQyRi6?p=preview
2. Click the `Submit (inside the form)` button on the bottom.
_(In this case the alert is shown but after dismissing it the UI remains inaccessible - visible but not usable.)_
3. Refresh the page (because the UI remains frozen after the previous step).
4. Enter invalid value in the textbox (shorter than 5 chars) and hit `Enter`.
_(In this case the alert is not even shown but the UI becomes inaccessible.)_

> **NOTE:** Notice that when clicking the top `Submit` button (the one in the navbar) or the secondary submit button (`Submit (outside of the form)`) everything works fine.

**Ionic Version**: 2.x

**Browser & Operating System**: Chrome

**Run `ionic info` from terminal/cmd prompt:**

```
Your system information:

Cordova CLI: 6.0.0
Ionic Version: 2.0.0-beta.3
Ionic CLI Version: 2.0.0-beta.19
Ionic App Lib Version: 2.0.0-beta.9
OS: Windows 8.1
Node Version: v5.8.0
```

**Comment by adamdbradley** [comment]
In your form you have both `<form [ngFormModel]="form" (ngSubmit)="submit(form.value)">` and the button has `<button list-item primary block (click)="submit(form.value)">`, so `submit(form.value)` is getting called twice, which opens up two alerts immediately causing the issue. When I remove the submit() from the button it then works. Does that solve the issue?

**Comment by iignatov** [comment]
Thanks for the great explanation!

Sorry, my bad, indeed there's a mistake in my code. I added the submit button in the form afterwards by copying the code from the button in the header and I didn't consider that it's the default form button and its handler gets called automatically when the form is submitted. **Removing the click handler from the form's submit button solves the issue.**

The second issue remains though, i.e. when using the `Enter` button to submit the form. The UI is not blocked anymore, but the alert is not displayed properly. Since it has little to do with the original issue I opened a new issue to track it (#5821) and I'm closing this one.
